Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.cpp

```diff
@@ -329,6 +331,36 @@ namespace AZ
             AzFramework::StringFunc::Path::GetFullFileName(request.m_fullPath.c_str(), destFileName);
             AzFramework::StringFunc::Path::ConstructFull(request.m_tempDirPath.c_str(), destFileName.c_str(), destPath, true);
 
+            // --- Ensure the BindPassSrg - flag is set if the pass-data has a PipelineViewTag set
+            auto& passTemplate{ passAsset.GetPassTemplate() };
+            if (passTemplate && passTemplate->m_passData)
+            {
+                auto* passData = azrtti_cast<RenderPassData*>(passTemplate->m_passData.get());
+                if (passData && !passData->m_pipelineViewTag.IsEmpty())
+                {
+                    // "PipelineViewTag": "MainCamera" is deprecated, discard the view-tag and set BindViewSrg to true
+                    if (passData->m_pipelineViewTag == AZ::Name("MainCamera"))
+                    {
+                        AZ_Warning(
+                            PassBuilderName,
+                            false,
+                            "Asset %s: '\"PipelineViewTag\": \"MainCamera\"' is deprecated, use '\"BindViewSrg\": true' instead",
+                            request.m_fullPath.c_str());
+                        passData->m_pipelineViewTag = AZ::Name();
+                    }
+                    else if (!passData->m_bindViewSrg)
+                    {
+                        // Explicitly set "PipelineViewTag": implicitly set BindViewSrg to true as well, if it isn't yet.
+                        AZ_Info(
+                            PassBuilderName,
+                            "Asset %s: Pass has explicit PipelineViewTag '%s' -> setting \"BindViewSrg\" to true as well.",
+                            request.m_fullPath.c_str(),
+                            passData->m_pipelineViewTag.GetCStr());
+                        passData->m_bindViewSrg = true;
+                    }
```

Comment:
The logic here doesn't set m_bindViewSrg to true if  m_pipelineViewTag was set to "MainCamera" for RenderPass.
Is this correct?

Reply:
good catch. Fixed it.